xds/src/main/java/io/grpc/xds/Filter.java

```diff
@@ -103,6 +119,36 @@ default ServerInterceptor buildServerInterceptor(
     return null;
   }
 
+  /**
+   * Implement to perform cleanup on filter shutdown.
+   *
+   * <p>Stateful filters should implement this method to perform cleanup, f.e. release shared
+   * resources, close remote connections, etc.
+   *
+   * <p>This method is called on the filter instance when an LDS update does not have
```

Comment:
I like there's documentation, but it seems wrong to be _here_. None of this influences the implementation of the method or the caller of the method, and it seems likely it'll become out-of-date since the code to make this happen is quite far away.

I guess I could also argue that it isn't right to focus on _shutting down_. It is shut down when no longer needed; the "filter instances are created per combination of" above is basically the same thing as here. The additional explanations here are meaningless to filters.

The newInstance() could still get out-of-date, but is a bit less brittle to change, and it is more useful as it tells filters the scope of their state.